lms/djangoapps/certificates/tests/test_signals.py

```diff
@@ -286,3 +288,44 @@ def test_cert_generation_on_photo_verification_instructor_paced(self):
                         'expected_verification_status': SoftwareSecurePhotoVerification.STATUS.approved
                     }
                 )
+
+
+@ddt.ddt
+class CertificateGenerationTaskTest(ModuleStoreTestCase):
+    """
+    Tests for certificate generation task.
+    """
+
+    def setUp(self):
+        super(CertificateGenerationTaskTest, self).setUp()
+        self.course = CourseFactory.create()
+
+    @ddt.data(
+        ('professional', True),
+        ('verified', True),
+        ('no-id-professional', True),
+        ('credit', True),
+        ('audit', False),
+        ('honor', False),
```

Comment:
why is `generate_certificate` task being created for `honor`, `audit` modes? 
I guess these are not in `allowed_enrollment_modes_list`?

Reply:
This test essentially verify that `generate_certificate` task is not being created for `honor` and `audit`. We call `fire_ungenerated_certificate_task` method which handles the logic around task creation with this list of enrollments  and assert that task is fired expectedly.